### DISCUSSION TITLE
Trying to use ScriptDOM

### DISCUSSION THREAD

**Discussion by moraja** [microsoft/DacFx#441] [ScriptDOM]
I'm looking for help identifying how I can parse a query like this one :
**declare @P0001 nvarchar(73)  set @P0001 = N'UPDATE session SET ss_last_used = @LastUsed WHERE ss_sequence = @Sequence'  declare @P0002 nvarchar(32)  set @P0002 = N'@LastUsed datetime,@Sequence int'  declare @P0003 datetime  set @P0003 = '20240305 15:53:09.093'  declare @P0004 int  set @P0004 = 9482  exec sp_executesql @P0001, @P0002, @LastUsed = @P0003, @Sequence = @P0004**

to extract the statement being executed , I'm wondering if there is a direct way to extract this info without having to inspect the code.: **"UPDATE session SET ss_last_used = @LastUsed WHERE ss_sequence = @Sequence "**

**Comment by dzsquared** [comment]
Hi @moraja - I put together a sample for this scenario - https://github.com/dzsquared/scriptdom-parse-sample

There's both the ability to get the initial query with parameters and the query with the values substituted.
